src/components/KanbanBoard.tsx

```diff
@@ -31,11 +77,50 @@ export function KanbanBoard() {
         }
     }
 
+    const updateTaskStatus = async (task: Task, endStatus: "Todo" | "In Progress" | "Completed" | null | undefined) => {
+        const { data, error, response } = await client.PATCH(`/tasks/{id}`, 
+        {   params: {
+                path: {
+                    id: task.id || ""
+                }
+            },
+            body: {
+                "name": task.name,
+                "description": task.description,
+                "dueDate": task.dueDate,
+                "priority": task.priority,
+                "status": endStatus,
+                "assignedTo": task.assignedTo,
+                "projectId": task.projectId,
+                "milestoneId": task.milestoneId,
+                "qaTask": {
+                    "name": task.qaTask.name,
+                    "description": task.qaTask.description,
+                    "dueDate": task.qaTask.dueDate,
+                    "priority": task.qaTask.priority,
+                    "status": task.qaTask.status,
+                    "assignedTo": task.qaTask.assignedTo
+                },
+                "dependentMilestones": task.dependentMilestones,
+                "dependentTasks": task.dependentTasks
+            }
+        });
```

Comment:
The way I implemented update task you should only need to send the things that changed so if you want you could just send `{"status": task.status, "qaTask": {"status": task.qaTask.status}}`.

But I think we might need to reuse this logic later for the edit task anyways so up to you.